activity_browser/ui/tables/models/base.py

```diff
@@ -420,18 +420,24 @@ class ABSortProxyModel(QSortFilterProxyModel):
 
     See this for context: https://github.com/LCA-ActivityBrowser/activity-browser/pull/1151
     """
-    def lessThan(self, left, right) -> bool:
+    def lessThan(self, left: QModelIndex, right: QModelIndex) -> bool:
         """Override to sort actual data, expects `left` and `right` are comparable.
 
         If `left` and `right` are not the same type, we check if numerical and empty string are compared, if that is the
         case, we assume empty string == 0.
         Added this case for: https://github.com/LCA-ActivityBrowser/activity-browser/issues/1215"""
-        left_data = self.sourceModel().data(left, 'sorting')
-        right_data = self.sourceModel().data(right, 'sorting')
+        left_data = self.sourceModel().data(left)
+        right_data = self.sourceModel().data(right)
+        if not left_data and not right_data:
+            return False
+
         if type(left_data) is type(right_data):
             return left_data < right_data
-        elif type(left_data) in (int, float) and type(right_data) is str and right_data == "":
+
+        if type(left_data) in (int, float) and not right_data:
             return left_data < 0
-        elif type(right_data) in (int, float) and type(left_data) is str and left_data == "":
+
+        if type(right_data) in (int, float) and not left_data:
             return 0 < right_data
+
         raise ValueError(f"Cannot compare {left_data} and {right_data}, incompatible types.")
```

Comment:
I found a missing test case that wasn't in the original code, thanks!

Please also add testing between `str` and `None` in addition to numeric and `None`. Example in UI: 2 exchanges in an activity, one with a comment and one without.